packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -184,6 +184,10 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     }
   }
 
+  public dismissMenu(): void {
+    this.setState({ menuProps: null });
+  }
```

Comment:
This should only set the state if it is changing. We don't want this to cause a rerender if someone calls this if nothing has changed

Reply:
Since state should could be updated async, it is generally a bad idea to use the previous state as a mechanism for computing the next state.

The optimization for skipping the render should be made using the "shouldComponentUpdate" lifecycle method instead. See https://facebook.github.io/react/docs/state-and-lifecycle.html for more info. 

Implementing shouldComponentUpdate is a nontrivial change that could cause some regressions, so I'm leaning towards ignoring that potential wasted render.

Reply:
I thought about that, but thought adding shouldComponentUpdate would be too heavy. I don't feel strongly either way if this happens to cause a wasted render, just thought I'd mention it